lib/auth/join_gcp.go

```diff
@@ -66,6 +66,9 @@ func checkGCPAllowRules(token *types.ProvisionTokenV2, claims *gcp.IDTokenClaims
 		if !slices.Contains(rule.ProjectIDs, compute.ProjectID) {
 			continue
 		}
+
+		// compute.Zone is not set if running in a GKE cluster under Workload Identity.
+		// In this case, `google` claim is not present and we can't check the zone.
 		if len(rule.Locations) > 0 && !slices.ContainsFunc(rule.Locations, func(location string) bool {
```

Comment:
What should we do here? 
`Locations` are never available without the `google` claim, so Pods won't be able to join if the location is specified in Teleport Token.

Should we skip the check if `compute.Zone==""`  or should we return an error saying: service account <ABC> didn't provide the location. Remove it from the token immediately

Reply:
I'm leaning towards emitting a warning here and rejecting the join - it's easy enough for them to modify the token to add an allow rule that doesn't specify a zone and with rules around joining/authz, it seems better to lean towards being too conservative.